src/main.cpp

```diff
@@ -1632,12 +1628,14 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
             const CCoins *coins = inputs.AccessCoins(prevout.hash);
             assert(coins);
 
-            // If prev is coinbase, check that it's matured
             if (coins->IsCoinBase()) {
-                if (nSpendHeight - coins->nHeight < COINBASE_MATURITY)
+                // Ensure that coinbases cannot be spent to transparent outputs
+                // Disabled on regtest
+                if (consensusParams.coinbaseMustBeProtected && !tx.vout.empty()) {
```

Comment:
Wait, no, that doesn't matter, because those fees get included in the coinbase of that mined block. So the only ways a coinbase can be spent is transparently as miner fees to a later block, or to a JoinSplit.